app/code/Transoft/Blog/Block/SingleBlog.php

```diff
@@ -0,0 +1,74 @@
+<?php
+declare(strict_types=1);
+
+namespace Transoft\Blog\Block;
+
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+use Transoft\Blog\Api\Data\BlogInterface;
+use Transoft\Blog\Model\BlogRepository;
+
+/**
+ * Generates blog list
+ */
+class SingleBlog extends Template
+{
+    /**
+     * @var RequestInterface $request
+     */
+    private $request;
+
+    /**
+     * @var BlogRepository
+     */
+    private $blogRepository;
+
+    /**
+     * Creates single blog block
+     *
+     * @param Context $context
+     * @param RequestInterface $request
+     * @param BlogRepository $blogRepository
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        RequestInterface $request,
+        BlogRepository $blogRepository,
+        array $data = []
+    ) {
+        $this->request = $request;
```

Comment:
Remove field. Use $this->_request (\Magento\Framework\View\Element\AbstractBlock::$_request) instead